### PR TITLE
Bug 1794142: Fixes unnecessary wait when readinessindicatorfile is not present

### PR DESCRIPTION
**Pull request by dougbtv**

### CODE REVIEW

File: multus/multus.go [hunk 1]
```diff
@@ -371,15 +371,19 @@ func cmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) (cn
...
-		_, err := os.Stat(n.ReadinessIndicatorFile)
-		switch {
-		case err == nil:
-			return true, nil
-		default:
-			return false, nil
-		}
-	})
+	if n.ReadinessIndicatorFile != "" {
+
```
Comment (danwinship) [inline multus/multus.go]: gratuitous blank line here and at the end

File: multus/multus.go [hunk 2]
```diff
@@ -371,15 +371,19 @@ func cmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) (cn
...
+		wait.ExponentialBackoff(defaultReadinessBackoff, func() (bool, error) {
+			_, err := os.Stat(n.ReadinessIndicatorFile)
+			switch {
+			case err == nil:
+				return true, nil
+			default:
+				return false, nil
+			}
```
Comment (danwinship) [inline multus/multus.go]: you could just do `return err == nil, nil`

File: multus/multus.go [hunk 3]
```diff
@@ -371,15 +371,19 @@ func cmdAdd(args *skel.CmdArgs, exec invoke.Exec, kubeClient k8s.KubeClient) (cn
...
-		switch {
-		case err == nil:
-			return true, nil
-		default:
-			return false, nil
-		}
-	})
+	if n.ReadinessIndicatorFile != "" {
+
+		wait.ExponentialBackoff(defaultReadinessBackoff, func() (bool, error) {
```
Comment (danwinship) [inline multus/multus.go]: Presumably you should be bailing out with an error if `wait.ExponentialBackoff` returns an error?

**Comment by dougbtv** [comment]
/retest

Looks like a flake, an error pulling an image

File: multus/multus.go (hunk 3 above)
Comment (s1061123) [reply multus/multus.go]: ExponentialBackoff returns error only if conditional function returns error.
In this case, we always returns nil at line 377, hence we don't have to.

https://godoc.org/k8s.io/apimachinery/pkg/util/wait#ExponentialBackoff

**Comment by dougbtv** [comment]
Note: I did also opt to include dcbw's commit for improved error handling during CNI ADD/DEL in order to use the standardized error logging to match upstream in order to address the error handling for the exponential backoff (thanks for the eye on that)

**Comment by dougbtv** [comment]
/retest

Looks like a flake with `received unexpected HTTP status: 504 Gateway Time-out`

**Pull request merged**